src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -377,7 +360,14 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
                     }
                     catch (Exception ex)
                     {
-                        exceptions.AddRange(groupedTasks.Exception?.InnerExceptions ?? new[] { ex }.AsEnumerable());
+                        if (groupedTasks.IsFaulted)
```

Comment:
Why was this changed?

Reply:
To reduce allocations.

Reply:
Sure; I was thinking of the new LOC for checking `IsFaulted` - when will it be `false`?